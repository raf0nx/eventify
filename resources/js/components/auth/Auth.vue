<template>
	<v-main>
		<v-container class="fill-height py-0" fluid>
			<v-row class="fill-height">
				<v-col cols="12" md="7" class="pa-0 auth-panel">
					<div
						class="auth-panel__container pa-4 d-flex flex-column justify-space-between"
					>
						<div class="d-flex logo">
							<v-icon class="white--text mr-1">mdi-buffer</v-icon>
							<span class="logo__text white--text font-weight-bold"
								>Eventify</span
							>
						</div>
						<div class="d-flex flex-column align-center mb-16">
							<h2 class="auth-panel__title white--text">
								Welcome back!
							</h2>
							<span
								class="auth-panel__subtitle grey--text text--lighten-3 mt-4"
								>Sign In to continue using my application!</span
							>
						</div>
						<div class="d-flex justify-center">
							<span class="white--text font-weight-bold"
								>v. 0.0</span
							>
						</div>
					</div>
				</v-col>
				<transition name="scale" mode="out-in">
					<router-view />
				</transition>
			</v-row>
		</v-container>
	</v-main>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	@Component
	export default class Auth extends Vue {}
</script>

<style>
	.auth-panel {
		background-image: url("https://i.pinimg.com/originals/c4/6f/3c/c46f3cfbff1b198e0276e4d577a4464c.png");
		background-position: center;
		background-repeat: no-repeat;
		background-size: cover;
	}

	.auth-panel__container {
		height: 100%;
	}

	.auth-panel__title {
		font-size: 4rem;
	}

	.auth-panel__subtitle {
		font-size: 1.5rem;
	}

	.logo__text {
		font-size: 1.5rem;
	}

	.auth-form__title {
		font-size: 3rem;
	}
</style>