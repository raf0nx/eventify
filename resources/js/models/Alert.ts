import { EnumAlertType } from "@/enums/EnumAlertType";

export class AlertModel {
    showAlert = false;
    type = EnumAlertType.SUCCESS;
    message = "";
    dismissible = false;
    callback: Function | null = null;
    btnText: string = "";

    setShowAlert(showAlert: boolean): AlertModel {
        this.showAlert = showAlert;
        return this;
    }

    setType(type: EnumAlertType): AlertModel {
        this.type = type;
        return this;
    }

    setMessage(message: string): AlertModel {
        this.message = message;
        return this;
    }

    setDismissible(dismissible: boolean): AlertModel {
        this.dismissible = dismissible;
        return this;
    }

    setCallback(callback: Function): AlertModel {
        this.callback = callback;
        return this;
    }

    setBtnText(text: string): AlertModel {
        this.btnText = text;
        return this;
    }
}
