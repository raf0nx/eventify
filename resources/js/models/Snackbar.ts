import { EnumSnackbarColor } from "@/enums/EnumSnackbarColor";
import { EnumSnackbarIcon } from "@/enums/EnumSnackbarIcon";

export class SnackbarModel {
    showSnackbar = false;
    message = "";
    icon = EnumSnackbarIcon.SUCCESS;
    color = EnumSnackbarColor.SUCCESS;

    setShowSnackbar(value: boolean): SnackbarModel {
        this.showSnackbar = value;
        return this;
    }

    setMessage(message: string): SnackbarModel {
        this.message = message;
        return this;
    }

    setIcon(icon: EnumSnackbarIcon): SnackbarModel {
        this.icon = icon;
        return this;
    }

    setColor(color: EnumSnackbarColor): SnackbarModel {
        this.color = color;
        return this;
    }
}
