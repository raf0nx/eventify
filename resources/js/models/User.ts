export interface User {
    id: Number;
    name: String;
    email: string;
    created_at: Date;
    email_verified_at?: Date;
    updated_at: Date;
}
