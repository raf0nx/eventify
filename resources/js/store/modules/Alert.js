export const namespaced = true;

export const state = {
    message: null,
    showAlert: false
};

export const mutations = {
    SET_ALERT(state, message) {
        state.showAlert = true;
        state.message = message;
    }
};

export const actions = {};

export const getters = {
    message: state => {
        return state.message;
    },
    showAlert: state => {
        return state.showAlert;
    }
};
