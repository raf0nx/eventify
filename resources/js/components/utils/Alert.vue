<template>
	<v-alert
		v-if="alert.showAlert"
		class="mb-0 font-weight-bold text-center"
		data-cy="alert"
		border="bottom"
		tile
		dense
		:type="alert.type"
		:dismissible="alert.dismissible"
	>
		{{ alert.message }}
		<v-btn
			v-if="alert.callback"
			@click="alert.callback"
			outlined
			class="ml-4 font-weight-bold"
			small
		>
			{{ alert.btnText }}
		</v-btn>
	</v-alert>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { AlertModel } from "@/models/Alert";

	@Component
	export default class Alert extends Vue {
		get alert(): AlertModel {
			return UtilsModule.alert;
		}
	}
</script>

<style scoped>
</style>