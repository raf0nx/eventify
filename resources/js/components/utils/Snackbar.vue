<template>
	<v-snackbar
		v-model="snackbar.showSnackbar"
        class="mb-8"
		app
		fixed
		bottom
		:color="snackbar.color"
	>
		<div class="d-flex align-center">
			<v-icon color="white">{{ snackbar.icon }}</v-icon>
			<span class="font-weight-bold ml-2">{{ snackbar.message }}</span>
		</div>
	</v-snackbar>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { SnackbarModel } from "@/models/Snackbar";

	@Component
	export default class Snackbar extends Vue {
		get snackbar(): SnackbarModel {
			return UtilsModule.snackbar;
		}
	}
</script>

<style scoped>
</style>