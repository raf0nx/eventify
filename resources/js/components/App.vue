<template>
	<v-app>
		<v-system-bar
			v-if="alert"
			color="orange darken-2"
			class="font-weight-bold justify-center white--text"
			data-cy="system_bar"
			app
			window
		>
			<v-icon class="white--text">mdi-alert</v-icon>
			Email not verified! Check your inbox at {{ userEmail }}
			<v-btn
				@click="resendVerificationLink()"
				plain
				text
				class="white--text font-weight-bold"
				small
			>
				Resend link
			</v-btn>
		</v-system-bar>
		<v-snackbar
			v-model="snackbar.showSnackbar"
			fixed
			bottom
			color="deep-purple"
			class="mb-12"
		>
			<div class="d-flex align-center">
				<v-icon color="white">mdi-check-circle</v-icon>
				<span class="font-weight-bold ml-1">{{
					snackbar.message
				}}</span>
			</div>
		</v-snackbar>
		<v-overlay :value="loader">
			<v-progress-circular indeterminate size="64"></v-progress-circular>
		</v-overlay>
		<transition name="scale" mode="out-in" appear>
			<router-view></router-view>
		</transition>
	</v-app>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import AuthService from "@/services/AuthService";
	import { AuthModule } from "@modules/Auth";
	import { Snackbar, UtilsModule } from "@modules/Utils";
	import { User } from "@/models/User";

	@Component
	export default class App extends Vue {
		get authUser(): User | null {
			return AuthModule.authUser;
		}

		get snackbar(): Snackbar {
			return UtilsModule.snackbar;
		}

		get userEmail(): string | null {
			return this.authUser ? this.authUser.email : null;
		}

		get alert(): boolean {
			return UtilsModule.alert;
		}

		get loader(): boolean {
			return UtilsModule.loader;
		}

		get userId(): Number | null {
			return this.authUser ? this.authUser.id : null;
		}

		async resendVerificationLink(): Promise<void> {
			UtilsModule.setLoading(true);
			try {
				await AuthService.sendVerification(this.userId);
				UtilsModule.setSnackbar({
					showSnackbar: true,
					message: "Verification Link resend successfully!",
				});
			} catch (error: unknown) {
				console.log(error);
			}
			UtilsModule.setLoading(false);
		}
	}
</script>

<style>
	.scale-enter-active,
	.scale-leave-active {
		transition: all 0.5s ease;
	}

	.scale-enter,
	.scale-leave-to {
		opacity: 0;
		transform: scale(0.9);
	}
</style>