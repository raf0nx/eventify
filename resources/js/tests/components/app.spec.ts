import Vuetify from "vuetify";
import VueRouter from "vue-router";
import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import axios from "axios";

import App from "@components/App.vue";
import { UtilsModule } from "@/store/modules/Utils";
import { AlertModel } from "@/models/Alert";
import { SnackbarModel } from "@/models/Snackbar";
import { AuthModule } from "@/store/modules/Auth";
import { user } from "@/tests/constans/User";

const EMAIL_NOT_VERIFIED = "Email not verified!";
const INBOX_CHECK = "Check your inbox at ";
const RESEND_LINK_MSG = "Verification Link resend successfully!";

jest.mock("axios");

describe("App.vue", () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);

    const router = new VueRouter();
    let vuetify: Vuetify;
    let wrapper: Wrapper<App>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(App, {
            localVue,
            vuetify,
            router,
            stubs: {
                Navbar: true,
                NavbarDrawer: true
            }
        });

        UtilsModule.setAlert(
            new AlertModel()
                .setShowAlert(true)
                .setMessage(EMAIL_NOT_VERIFIED + INBOX_CHECK + user.email)
                .setCallback(() => true)
                .setBtnText("Send")
        );

        UtilsModule.setSnackbar(
            new SnackbarModel()
                .setShowSnackbar(true)
                .setMessage(RESEND_LINK_MSG)
        );
    });

    afterEach(() => {
        wrapper.destroy();
        jest.clearAllMocks();
    });

    it("Should match the snapshot", () => {
        // Assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should get authenticated user", async () => {
        // Assert
        // @ts-ignore
        axios.get.mockImplementationOnce(() => Promise.resolve({ data: user }));

        // Act
        const authUser = await AuthModule.getAuthUser();

        // Assert
        // @ts-ignore
        expect(wrapper.vm.user).toEqual(authUser);
    });

    it("Should not get authenticated user", async () => {
        // Assert
        // @ts-ignore
        axios.get.mockImplementationOnce(() => Promise.reject());

        // Act
        await AuthModule.getAuthUser();

        // Assert
        // @ts-ignore
        expect(wrapper.vm.user).toBeNull();
    });

    it("Should render alert and its content", () => {
        // Arrange
        wrapper = mount(App, {
            localVue,
            vuetify,
            router,
            computed: {
                user: () => true
            },
            stubs: {
                Navbar: true,
                NavbarDrawer: true
            }
        });

        // Act
        const alert = wrapper.find(".v-alert");
        const button = wrapper.find("button");

        // Assert
        expect(alert.text()).toContain(
            EMAIL_NOT_VERIFIED + INBOX_CHECK + user.email
        );
        expect(button.text()).toBe("Send");
    });

    it("Should render snackbar and it's message", () => {
        // Arrange
        wrapper = mount(App, {
            localVue,
            vuetify,
            router,
            computed: {
                user: () => true
            },
            stubs: {
                Navbar: true,
                NavbarDrawer: true
            }
        });
        const snackbar = wrapper.find(".v-snack");

        // Assert
        expect(snackbar.text()).toContain(RESEND_LINK_MSG);
    });
});
