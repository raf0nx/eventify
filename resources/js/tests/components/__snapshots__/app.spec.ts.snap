// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Should match the snapshot 1`] = `
"<div data-app=\\"true\\" class=\\"v-application v-application--is-ltr theme--light\\" id=\\"app\\">
  <div class=\\"v-application--wrap\\">
    <!---->
    <div class=\\"v-overlay theme--dark\\" style=\\"z-index: 5;\\">
      <div class=\\"v-overlay__scrim\\" style=\\"opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;\\"></div>
    </div>
    <transition-stub name=\\"scale\\" appear=\\"true\\" mode=\\"out-in\\">
      <!---->
    </transition-stub>
  </div>
</div>"
`;
